scipy/stats/_multivariate.py

```diff
@@ -2648,6 +2661,394 @@ def entropy(self):
         method.__doc__, wishart_docdict_noparams)
     method.__doc__ = doccer.docformat(method.__doc__, wishart_docdict_params)
 
+_multinomial_doc_default_callparams = """\
+n : int
+    Number of trials
+p : array_like
+    Probability of a trial falling into each category; should sum to 1
+"""
+
+_multinomial_doc_callparams_note = \
+"""`n` should be a positive integer. Each element of `p` should be in the
+interval :math:`[0,1]` and the elements should sum to 1. If they do not sum to
+1, the last element of the `p` array is not used and instead interpreted as the
+remaining probability left over from the earlier elements.
+"""
+
+_multinomial_doc_frozen_callparams = ""
+
+_multinomial_doc_frozen_callparams_note = \
+    """See class definition for a detailed description of parameters."""
+
+multinomial_docdict_params = {
+    '_doc_default_callparams': _multinomial_doc_default_callparams,
+    '_doc_callparams_note': _multinomial_doc_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+multinomial_docdict_noparams = {
+    '_doc_default_callparams': _multinomial_doc_frozen_callparams,
+    '_doc_callparams_note': _multinomial_doc_frozen_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+class multinomial_gen(multi_rv_generic):
+    r"""
+    A multinomial random variable.
+
+    .. versionadded:: 0.18.0
+
+    Methods
+    -------
+    ``pmf(x, n, p)``
+        Probability mass function.
+    ``logpmf(x, n, p)``
+        Log of the probability mass function.
+    ``rvs(n, p, size=1, random_state=None)``
+        Draw random samples from a multinomial distribution.
+    ``entropy(n, p)``
+        Compute the entropy of the multinomial distribution.
+    ``cov(n, p)``
+        Compute the covariance matrix of the multinomial distribution.
+
+    Parameters
+    ----------
+    x : array_like
+        Quantiles, with the last axis of `x` denoting the components.
+    %(_doc_default_callparams)s
+    %(_doc_random_state)s
+
+    Notes
+    -----
+    %(_doc_callparams_note)s
+
+    Alternatively, the object may be called (as a function) to fix the `n` and
+    `p` parameters, returning a "frozen" multinomial random variable:
+
+    The probability mass function for `multinomial` is
+
+    .. math::
+
+        f(x) = \begin{pmatrix} n \\ x_1, \ldots, x_k \end{pmatrix}
+        p_1^{x_1} \cdots p_k^{x_k},
+
+    supported on :math:`x=(x_1, \ldots, x_k)` where each :math:`x_i` is a
+    nonnegative integer and their sum is :math:`n`.
```

Comment:
The current implementation accepts non-integer `x_i`, does it? 
(I'm not saying it should, all I'm saying that the docstring should match the implementation.)